### PR TITLE
Add relational diagram for CRs

### PR DESCRIPTION
**Pull request by fmuyassarov**
Add:

- diagram showing how CRs are referenced
- diagram showing how objects are referenced

### CODE REVIEW

**Comment by russellb** [comment]
/approve

looks useful!  I'd like someone more familiar with v1alpha2 to review this and give a "lgtm"

**Comment by kanwar-saad** [comment]
@maelk second CR diagram is useful in the sense that it describes what user configures. Owner references and status data is added by controllers.That should be documented in separate diagrams.

File: crs_reference.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+This document explains relationship between all the Custom Resources (CRs)
+required to create the target Kubernetes cluster on bare metal machine and how
+they reference to each other.
+
+## Environment Variables
+
+The user is required to set the following environment variables before applying
```
Comment (kanwar-saad) [inline crs_reference.md]: can you please explain that which CRs are you talking about and from which repository .. is it from the operators or metal3 dev env e.t.c.

File: images/cr_references.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
...5)rotate(-30 551.5 338.07179676972453)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe flex-start; justify-content: unsafe center; width: 1px; height: 1px; padding-top: 338px; margin-left: 552px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: nowrap; "><font style="font-size: 12px" color="#006666">consumerRef</font></div></div></div></foreignObject><text x="552" y="350" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">consumerRef</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://desk.draw.io/support/solutions/articles/16000042487" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (maelk) [inline images/cr_references.svg]: The infraref for machine/BMMachine is the wrong way around. And there is also an ownerref from BMMachine to Machine

File: crs_reference.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+This document explains relationship between all the Custom Resources (CRs) required to create the target Kubernetes cluster on bare metal machine and how they reference to each other. To see the example CRs check [cluster-api-provider-baremetal](https://github.com/metal3-io/cluster-api-provider-baremetal/tree/release-0.2/examples)
```
Comment (Xenwar) [inline crs_reference.md]: This document explains the relationship

File: crs_reference.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+IMAGE_URL
+KUBERNETES_VERSION
+SSH_PUB_KEY_CONTENT
+
+```
+
+### Cluster and Machine
+
+##### CAPI v1alpha2
+This diagram describes object relationship based on CAPI v1alpha2 and how they reference to each other.
```
Comment (Xenwar) [inline crs_reference.md]: objects'

File: crs_reference.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+![crs](images/v1a2_crs.svg)
+
+##### CAPI v1alpha3
+This diagram describes object relationship based on CAPI v1alpha3 and how they reference to each other.
```
Comment (Xenwar) [inline crs_reference.md]: objects'

**Comment by fmuyassarov** [comment]
> We just removed clusterName from BM* in v1alpha3. It was unneeded

I have removed v1alpha3 content here. So, for now this is only for v1alpha2.

**Comment by maelk** [comment]
I am confused, there are v1alpha3 crs there

**Comment by fmuyassarov** [comment]
@maelk Sorry I missed to remove them.

**Pull request merged**